File: Platform/codesignal/Intro/11.isLucky/solution.js
function solution(n) {
  let half = (""+n).length / 2
  return (""+n).slice(0, half).split("").map(x => +x).reduce((a,b) => (a+b), 0) === (""+n).slice(half).split("").map(x => +x).reduce((a,b) => (a+b), 0)
}

function solution2(n) {
  let arr = ("" + n).split("");
  let result = 0;
  for (let i = 0; i < arr.length / 2; i++) {
    result += +arr[i] - arr[arr.length - 1 - i];
  }
  return result === 0 ? true : false;
}

console.log(solution(1230));

File: Platform/codesignal/Intro/11.isLucky/solution.py
def solution(n):
  half = len(str(n))//2
  return sum([int(x) for x in str(n)[:half]]) == sum([int(x) for x in str(n)[half:]])


def solution2(n):
  result = 0
  s = str(n)
  for i in range(len(s) // 2):
    result += (int(s[i]) - int(s[-1-i]))
  return result == 0

print(solution(1230))